### PR TITLE
Fetch payload dynamically in request/response details

### PR DESCRIPTION
**Pull request by nsrCodes**
builds on top of changes here - https://github.com/requestly/requestly-cloud/pull/399

### CODE REVIEW

File: app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx [hunk 1]
```diff
@@ -1,41 +1,70 @@
...
 import { ThemeProvider } from "@devtools-ds/themes";
 import { ObjectInspector } from "@devtools-ds/object-inspector";
 import { NetworkLogProperty } from "./NetworkLogProperty";
 
 type Props = {
   label?: string;
-  payload: any;
+  payload?: any;
   isPayloadTooLarge?: boolean;
+  fetchPayload?: () => Promise<any>;
```
Comment (rohanmathur91) [inline app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx:11]: `fetchPayload` this should not be exposed from `NetworkPayload`, ideally `NetworkPayload` should just take the payload, it should not know how to fetch any payload.

File: app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx [hunk 2]
```diff
@@ -1,41 +1,70 @@
...
+          } else {
+            setFinalPayload(fetchedPayload);
+          }
+        })
+        .catch((err) => {
+          console.error(err);
+        })
+        .finally(() => {
+          setFetching(false);
+        });
```
Comment (rohanmathur91) [inline app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx:46]: `RQNetworkTable` should know the currently selected log ie its parent component should know the selected log and there we should fetch the payload for the selected log and just update the logs state.

File: app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx [hunk 3]
```diff
@@ -1,41 +1,70 @@
...
-    return payload;
-  }, [payload]);
+  }, [payload, fetchPayload]);
 
-  if (typeof parsedPayload === "object") {
+  if (fetching) {
+    return <NetworkLogProperty label={label}>Loading...</NetworkLogProperty>;
+  }
+
+  if (typeof finalPayload === "object") {
```
Comment (rohanmathur91) [inline app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx:54]: Will `finalPayload` be ever `null`, if yes then we should add that check too.

**Review by rohanmathur91 — CHANGES_REQUESTED** [review]

File: app/src/lib/design-system/components/RQNetworkTable/index.tsx [hunk 4]
```diff
@@ -78,10 +80,40 @@ export const RQNetworkTable: React.FC<RQNetworkTableProps> = ({
     setActiveLogId(closestLog?.log?.id);
   }, [logs, sessionCurrentOffset, sessionRecordingStartTime]);
 
+  const finalLogs = useMemo(() => {
+    let finalLogs = logs;
```
Comment (rohanmathur91) [inline app/src/lib/design-system/components/RQNetworkTable/index.tsx:84]: Create a copy of logs, the above might be working now, but might cause some unexpected issues in future
```suggestion
    let finalLogs = [...logs];
```

**Review by rohanmathur91 — CHANGES_REQUESTED** [review]

File: app/src/lib/design-system/components/RQNetworkTable/index.tsx [hunk 5]
```diff
@@ -30,6 +31,7 @@ export const RQNetworkTable: React.FC<RQNetworkTableProps> = ({
   disableFilters = false,
 }) => {
   const [activeLogId, setActiveLogId] = useState(null);
+  const [selectedLog, setSelectedLog] = useState<RQNetworkLog | null>(null);
```
[the remark this answers is not in the document]
Comment (nsrCodes) [reply app/src/lib/design-system/components/RQNetworkTable/index.tsx:34]: no. `activeLogId` is the used for syncing with the time in session recording viewer

[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/lib/design-system/components/RQNetworkTable/index.tsx:35]: So can we give a more descriptive name?

**Review by rohanmathur91 — APPROVED** [review]

**Pull request merged**